src/execution/reservoir_sample.cpp

```diff
@@ -40,39 +45,79 @@ void ReservoirSample::AddToReservoir(DataChunk &input) {
 }
 
 unique_ptr<DataChunk> ReservoirSample::GetChunk() {
-	return reservoir.Fetch();
+	if (num_added_samples == 0) {
+		return nullptr;
+	}
+	if (reservoir_chunk->size() > STANDARD_VECTOR_SIZE) {
+		// get from the back
+		auto ret = make_uniq<DataChunk>();
+		auto samples_remaining = num_added_samples - STANDARD_VECTOR_SIZE;
+		auto reservoir_types = reservoir_chunk->GetTypes();
+		SelectionVector sel(STANDARD_VECTOR_SIZE);
+		for (idx_t i = samples_remaining; i < num_added_samples; i++) {
```

Comment:
It was the original functionality if I'm not mistaken. 
I could start scanning from the front. If I do that, wouldn't I need to initialize two selection vectors on the reservoir chunk?

One would return the first standard vector size chunk, and the other selection vector would replace the original reservoir chunk with the first Standard vector size of tuples missing.

Let me know if that thinking is correct, if so, I think we should just return from the back to avoid the extra selection vector complexity. I've added it as a comment for now